### PR TITLE
Add firebase stuff to login and registration backend

### PR DESCRIPTION
**Pull request by KellyPSun**

### CODE REVIEW

**Comment by OddJerbb** [comment]
Looks good to me! I would also wait for @jcderry5's feedback though.

**Review by OddJerbb — APPROVED** [review]

File: step-strikers-app/step-strikers-app/Classes/Caster.swift [hunk 1]
```diff
@@ -47,14 +47,11 @@ class Caster: RPGCharacter {
         targetRef.getDocument { (document, error) in
             if let document = document, document.exists {
                 // get info about target's current weapon
-                let weaponName = document.get("current_weapon") as! String
-                let weaponMap = document.get("weapon_inventory") as! [String:[String:Int]]
-                let condition = weaponMap[weaponName]!["condition"]
-                let useCount = weaponMap[weaponName]!["use_count"]
+                let weapon = document.get("current_weapon") as! String
```
Comment (jcderry5) [inline step-strikers-app/step-strikers-app/Classes/Caster.swift:51]: Can you add a TODO for me split the weapon into weapon and useCount?

Comment (KellyPSun) [reply step-strikers-app/step-strikers-app/Classes/Caster.swift:51]: Yup yup

File: step-strikers-app/step-strikers-app/LoginViewController.swift [hunk 2]
```diff
@@ -34,8 +40,65 @@ class LoginViewController: UIViewController {
...
+            print("username not entered")
+        } else if passwordTextField!.text == "" {
+            // TODO: @Nick don't print this, display it on the screen
+            print("password not entered")
+        } else {
+            // TODO: @Kelly validate password
+            let docRef = Firestore.firestore().collection("players").document(self.usernameTextField!.text!)
+            docRef.getDocument { (document, error) in if let document = document, document.exists {
+                // TODO: @jalyn you can remove all these prints after you're done. I used them to test and left them in case they're helpful!
+                let characterName = document.get("character_name") as! String
```
Comment (jcderry5) [inline step-strikers-app/step-strikers-app/LoginViewController.swift:56]: Two things:
Does firebase store the class of the character (e.g bard, fights, etc.) If so, I would need this.
Also need current spell points for the casters.

File: step-strikers-app/step-strikers-app/LoginViewController.swift [hunk 3]
```diff
@@ -34,8 +40,65 @@ class LoginViewController: UIViewController {
...
+                }
+                let currWeapon = document.get("current_weapon") as! String
+                print("disgusting currWeapon string: \(currWeapon)")
+                
+                let armorInventory = document.get("armor_inventory") as! [String]
+                for armor in armorInventory {
+                    print("disgusting armor string: \(armor)")
+
+                    // TODO: @jalyn create armor objects and add each weapon to the local inventory one by one
+                }
```
Comment (jcderry5) [inline step-strikers-app/step-strikers-app/LoginViewController.swift:79]: the three modifiers need to be pulled (attackModifier, defenseModifier, magicResistanceModifier)

**Review by jcderry5 — CHANGES_REQUESTED** [review]

**Review by jcderry5 — APPROVED** [review]
You've done my changes! Thanks!

**Pull request merged**